addons/mail/static/src/models/thread_cache/thread_cache.js

```diff
@@ -94,6 +94,17 @@ function factory(dependencies) {
         // Private
         //----------------------------------------------------------------------
 
+        /**
+         * @private
+         */
+        _computeCheckedMessages() {
+            if (this.checkedMessages) {
```

Comment:
You had the issue? x2many should default to `[]` normally and it should never be falsy.

Reply:
No, (too ?) defensive programming